app/models/ability.rb

```diff
@@ -464,7 +477,37 @@ def to_dataset_item(user, is_guest)
     end
 
     # actions any logged in user can access
-    can [:new, :index, :filter], DatasetItem
+    can [:new, :index, :filter, :filter_todo], DatasetItem
+
+  end
+
+
+  def to_progress_event(user, is_guest)
+
+    # anyone can create as long as they have read access on the ancestor project of the dataset item
+    can [:create], ProgressEvent do |progress_event|
+      check_model(progress_event)
+
+      begin
+        Access::Core.can_any?(user, :reader, progress_event.dataset_item.audio_recording.site.projects)
+      rescue
+        fail CustomErrors::UnprocessableEntityError.new('Invalid dataset item')
+        false
```

Comment:
why is this false here? At the time I was not sure if the fail would return from the method, so this ensures that permission is denied if this is the case. 
can you document why this try catch is required? 
this try-catch catches all types of errors: that is not ok.
It was there because the deeply nested call to projects failed if dataset item was invalid and therefore didn't have an audio recording associated. I have refactored it to explicitly check for audio recording and removed the try-catch

Reply:
cool. AFAIK that false never would have had any value. Even if the exception had been handled, that line would have been skipped. Also if we are returning values early in the method, it's better to use an explicit `return` keyword to remove ambiguity.

> I have refactored it to explicitly check for audio recording and removed the try-catch
Even better.